app/src/test/java/com/goldenraven/padawanwallet/data/RepositoryTest.kt

```diff
@@ -0,0 +1,130 @@
+package com.goldenraven.padawanwallet.data
+
+import android.content.SharedPreferences
+import com.goldenraven.padawanwallet.utils.RequiredInitialWalletData
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.*
+
+
+class RepositoryTest {
+
+    private val repo: Repository = Repository
+    private lateinit var sharedPrefsMock: SharedPreferences
+    private lateinit var editorMock: SharedPreferences.Editor
+
+    @Before
+    fun setUp() {
+        sharedPrefsMock = mock(SharedPreferences::class.java)
+        editorMock = mock(SharedPreferences.Editor::class.java)
+        repo.setSharedPreferences(sharedPrefsMock)
+        `when`(sharedPrefsMock.edit()).thenReturn(editorMock)
+    }
+
+    @Test
+    fun `Check if wallet exists`(){
```

Comment:
Small nit, but since you might go back and fix other things in this PR I figured I'd point it out: let's make sure we have a space between the `()` and the function body:
```kotlin
fun `Check if wallet exists`() {
}
```

This comment is valid for all functions below.